File: 프로그래머스/lv2/49993. 스킬트리/스킬트리.py
import re


def solution(skill: str, skill_trees: list) -> int:
    answer = 0
    
    for skill_tree in skill_trees:
        target = re.sub(f'[^{skill}]', '', skill_tree)
        if skill.startswith(target):
            answer += 1
    
    return answer

File: 프로그래머스/2/49993. 스킬트리/스킬트리.java
import java.util.Arrays;

class Solution {
    public int solution(String skill, String[] skill_trees) {
        return (int) Arrays.stream(skill_trees)
                .map(v -> v.replaceAll("[^" + skill + "]", ""))
                .filter(v -> skill.startsWith(v))
                .count();
    }
}